src/metabase/db/metadata_queries.clj

```diff
@@ -36,39 +36,47 @@
       :type/Date     [:> field-form "0001-01-01"]
       :type/DateTime [:> field-form "0001-01-01T00:00:00"])))
 
-(defn- query-with-default-partitioned-field-filter
-  [query table-id]
-  (let [;; In bigquery, range or datetime partitioned table can have only one partioned field,
-        ;; Ingestion time partitioned table can use either _PARTITIONDATE or _PARTITIONTIME as
-        ;; partitioned field
-        partition-field (or (t2/select-one :model/Field
-                                           :table_id table-id
-                                           :database_partitioned true
-                                           :active true
-                                           ;; prefer _PARTITIONDATE over _PARTITIONTIME for ingestion time query
-                                           {:order-by [[:name :asc]]})
-                            (throw (ex-info (format "No partitioned field found for table: %d" table-id)
-                                            {:table_id table-id})))
-        filter-form     (partition-field->filter-form partition-field)]
-    (update query :filter (fn [existing-filter]
-                            (if (some? existing-filter)
-                              [:and existing-filter filter-form]
-                              filter-form)))))
+(defn add-required-filter-if-needed
+  "Add a dummy filter for tables that require a filters.
```

Comment:
I think this name misleads a bit, because it can add multiple "filters"
```suggestion
(defn add-required-filters-if-needed
  "Add a dummy filters for tables that require filters.
```